source/Calamari.Tests/KubernetesFixtures/ResourceStatus/ResourceStatusCheckerTests.cs

```diff
@@ -0,0 +1,214 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.Kubernetes.Integration;
+using Calamari.Kubernetes.ResourceStatus;
+using Calamari.Kubernetes.ResourceStatus.Resources;
+using Calamari.Testing.Helpers;
+using FluentAssertions;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using NUnit.Framework.Constraints;
+
+namespace Calamari.Tests.KubernetesFixtures.ResourceStatus
+{
+    
+    [TestFixture]
+    public class ResourceStatusCheckerTests
+    {
+        [Test]
+        public void ShouldReportStatusesWithIncrementingCheckCount()
+        {
+            var retriever = new TestRetriever();
+            var reporter = new TestReporter();
+            var resourceStatusChecker = new ResourceStatusChecker(retriever, reporter, new InMemoryLog());
+
+            var timer = new TestTimer(5);
+
+            for (var i = 0; i < 5; i++)
+            {
+                retriever.SetResponses(new List<Resource>{ new TestResource("Pod", Kubernetes.ResourceStatus.Resources.ResourceStatus.InProgress) });
+            }
+            
+            resourceStatusChecker.CheckStatusUntilCompletionOrTimeout(
+                new ResourceIdentifier[]
+                {
+                    new ResourceIdentifier("Pod", "my-pod", "default")
+                }, timer, null, new Options());
+            
+            reporter.CheckCounts().Should().BeEquivalentTo(new List<int>() { 1, 2, 3, 4, 5 });
+        }
+
+        [Test]
+        public void SuccessfulBeforeTimeout_ShouldReturnAsSuccessful()
+        {
+            var retriever = new TestRetriever();
+            var reporter = new TestReporter();
+            var resourceStatusChecker = new ResourceStatusChecker(retriever, reporter, new InMemoryLog());
+
+            var timer = new TestTimer(5);
+
+            retriever.SetResponses(
+                new List<Resource> { new TestResource("Pod", Kubernetes.ResourceStatus.Resources.ResourceStatus.Successful) }
+            );
+            
+            var result = resourceStatusChecker.CheckStatusUntilCompletionOrTimeout(
+                new ResourceIdentifier[]
+                {
+                    new ResourceIdentifier("Pod", "my-pod", "default")
+                }, timer, null, new Options());
+
+            result.Should().Be(true);
+        }
+
+        [Test]
+        public void FailureBeforeTimeout_ShouldReturnAsFailed()
+        {
+            var retriever = new TestRetriever();
+            var reporter = new TestReporter();
+            var resourceStatusChecker = new ResourceStatusChecker(retriever, reporter, new InMemoryLog());
+
+            var timer = new TestTimer(5);
```

Comment:
Its not clear to me why some of these tests use `TestTimer(5)` and others `TestTimer(2)`
Perhaps more explicit timers would make this clearer.